docs/guides/environmental-variables.md

```diff
@@ -0,0 +1,57 @@
+
+To go about this, there are broadly two ways:
+
+1. **Hardcode the values in our code** \
+   This is perhaps the easiest approach to solve the problem, but the most dangerous (and inefficient) one. Once you hardcode the values in your codebase, everyone who works on the code will be able to see your sensitive information, which poses a massive security issue. Also, if you want to change the values, you will have to change the code and redeploy the application, which is not ideal.
+
+2. **Store the values in environmental variables** \
+   This is the best approach to solve the problem. You can store the sensitive values in the OS of the server running your application, and during the runtime of your application, you can access these values from the OS. All programming languages have excellent support for this. This way, you can keep your sensitive information safe, and also change the values without having to change anything in your code.
+
+## Environmental variables in Tailcall
+
+With Tailcall, you can seamlessly fit in environmental variables into your GraphQL schema. Tailcall supports this by providing a `env` [Context](context.md) variable. This Context is then shared across all the operators, which you will use to resolve values in your schema.
+
+For example, you can do the following:
+
+```graphql showLineNumbers
+type Query {
+  users: [User]! @http(baseUrl: "{{env.API_ENDPOINT}}", path: "/users")
+}
+```
+
+Here, the `API_ENDPOINT` is the environmental variable. This value needs to be already set in the device where your server will be running. The value gets picked up by the server when it starts up and is available in the `env` Context variable.
+
+## Security aspects and best practices
+
+Environmental variables help you in mitigating most of the security risks. While this is true, it is also necessary to understand that, it doesn't actually help you in removing the risks completely, since the values are still in plain text. Configurational values might not be as sensitive, but the secrets still do have a chance to be compromised.
+
+To make sure your secrets stay safe and secure all through, here are some tips and tricks that you might consider using for your application:
+
+- **Use a `.env` file** \
+  This is a common practice that is followed by most developers. You can create a `.env` file in the root directory of your project and store all your environmental variables in it. This file should not be committed to your version control system, and should be added to the `.gitignore` file. This way, you can make sure that your secrets are not exposed to the public. Additionally, you can also use a `.env.example` file to list out all the environmental variables that are required for your application to run. This way, you can make sure that all the developers who work on your project are aware of the variables that are required for the application to run.
```

Comment:
Firstly Tailcall can't read .env files just yet. Secondly the best practices could be more from tailcall's point of view. For when to use an env variable and when not to use an env variable.

Reply:
got it